src/ORM/DataObject.php

```diff
@@ -663,7 +664,8 @@ protected function duplicateHasManyRelation($sourceObject, $destinationObject, $
         $source = $sourceObject->getComponents($relation);
         $dest = $destinationObject->getComponents($relation);
 
-        foreach ($source as $item) {
+        // Disabling sort improves performance
+        foreach ($source->sort(null) as $item) {
```

Comment:
This loop just adds items to a `DataList`.
The order in which items are added doesn't matter.